src/combinatorics/simple_graph/connectivity.lean

```diff
@@ -1347,6 +1347,85 @@ by { rw ← set.nonempty_iff_univ_nonempty, exact hconn u v }
 lemma connected.set_univ_walk_nonempty (hconn : G.connected) (u v : V) :
   (set.univ : set (G.walk u v)).nonempty := hconn.preconnected.set_univ_walk_nonempty u v
 
+/-! ### Walks as subgraphs -/
+
+namespace walk
+variables {G}
+
+/-- The subgraph consisting of the vertices and edges of the walk. -/
+@[simp] protected def to_subgraph : Π {u v : V}, G.walk u v → G.subgraph
+| u _ nil := G.singleton_subgraph u
+| _ _ (cons h p) := G.subgraph_of_adj h ⊔ p.to_subgraph
+
+@[simp] lemma mem_verts_to_subgraph {u v : V} (p : G.walk u v) (w : V) :
+  w ∈ p.to_subgraph.verts ↔ w ∈ p.support :=
+begin
+  induction p,
+  { simp },
+  { have : w = p_v ∨ w ∈ p_p.support ↔ w ∈ p_p.support :=
+    ⟨by rintro (rfl | h); simp [*], by simp { contextual := tt}⟩,
+    simp [p_ih, or_assoc, this], },
+end
+
+lemma verts_to_subgraph {u v : V} (p : G.walk u v) :
+  p.to_subgraph.verts = {w | w ∈ p.support} :=
+set.ext (mem_verts_to_subgraph p)
+
+instance {u v : V} (p : G.walk u v) : nonempty p.to_subgraph.verts :=
+⟨⟨u, by simp only [mem_verts_to_subgraph, start_mem_support]⟩⟩
+
+@[simp] lemma mem_edges_to_subgraph {u v : V} (p : G.walk u v) (e : sym2 V) :
```

Comment:
Having `p` explicit is arguable for dot notation, but I would strongly expect `e` to be implicit in an iff.
```suggestion
@[simp] lemma mem_edges_to_subgraph {u v : V} (p : G.walk u v) {e : sym2 V} :
```